### PR TITLE
[source][condition]  Allow to filter Github Release per release type

### PR DESCRIPTION
**Pull request by olblak**
Allow to filter GitHub Release per release type (draft,prelease,release,latest)
I am also adding the missing Github Release condition

```
conditions:
  default:
    name: Default Get Latest Release
    kind: githubrelease
    sourceid: default
    spec:
      owner: updatecli
      repository: updatecli
      typefilter:
          # draft: true
          # prerelease: true
          # release: true
          latest: true
      token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'

```
## Test

To test this pull request, you can run the following commands:

```shell
go build -o bin/updatecli
./bin/updatecli diff --config e2e/updatecli.d/success.d/githubRelease.yaml --debug 
```

### Tradeoff

For historical reason https://github.com/updatecli/updatecli/pull/196
The Github Release fallback to searching Git Tag if no GitHub release could be found.
This behavior is incompatible with filter release by tag as it would return the latest git tag if no release could be found matching release type.

To not introduce breaking change, I am introducing the parameter `disabletagsearch` which can be set to true to enforce the new behavior. This parameter will be remove in a futur release once we are ready to remove the old behavior.

### CODE REVIEW

**Comment by olblak** [comment]
After thinking more about this PR.
I'll change a little bit the approach.
I am gonna remove the parameter `disabletagsearch`

We keep the current deprecate behavior only if no release type have been specified and we couldn't find a release.
All the other scenario just ignore gittag

**Comment by olblak** [comment]
As part @lemeurherve @dduportal feedback, I revert the last commit and changed the key from `type` to `typefilter`

File: e2e/updatecli.d/deprecated.d/githubRelease.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+title: Test source githubRelease
```
[the remark this answers is not in the document]
Comment (dduportal) [reply e2e/updatecli.d/deprecated.d/githubRelease.yaml:2]: but why? I mean I don't have a solution availble to replace it: gittag is not an acceptable replacement giving how heavy, slow and buggy due to updatecli local git caching

[the remark this answers is not in the document]
Comment (olblak) [reply e2e/updatecli.d/deprecated.d/githubRelease.yaml:2]: and I see the example is wrong

[the remark this answers is not in the document]
Comment (olblak) [reply e2e/updatecli.d/deprecated.d/githubRelease.yaml:2]: > Beside I do not really understand why deprecating this behavior adds any value to updatecli users?

While I find that behavior confusing, it doesn't affect my use-cases so I am fine to not deprecate it now. Using the `typefilter`, I can optout do 🤷🏾 

> gittag is not an acceptable replacement giving how heavy, slow and buggy due to updatecli local git caching

We already have most of the code for a `githubTag` plugin so I could consider adding that new plugin. While the gittag is slow on very large git repositories that I am working, I must admit it doesn't bother me enough to work on it but I can keep on eye on it.

[the remark this answers is not in the document]
Comment (dduportal) [reply e2e/updatecli.d/deprecated.d/githubRelease.yaml]: > We already have most of the code for a githubTag plugin

Make sense: if we are able to provide a "drop-in" replacement, then the depreciation message would make sense then.

[the remark this answers is not in the document]
Comment (olblak) [reply e2e/updatecli.d/deprecated.d/githubRelease.yaml]: I am removing the warning message for  now so we can move forward with the current PR, I'll open an issue to track what has to be done for the githubtag resource

**Comment by dduportal** [comment]
A quick comment to confirm that this PR introduces a regression to the GH releases filtering when using a regexp (at least) filter.

The manifest https://github.com/jenkinsci/docker-agent/blob/master/updatecli/updatecli.d/jdk11.yaml allows to reproduce the issue:
- Before this PR (updatecli 0.37.0 for instance), the returned JDK11 version from temurin is [`jdk-11.0.17+8`](https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.17%2B8) which was released the 25 of October.
- After this PR, the returned version is now https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.16.1%2B1 which is from the 19th of August.

I'm going to run a debuger to understand why is the regression.

**Comment by dduportal** [comment]
Ok, got the issue: it seems that the default values when `typeFilter` is not specified are incorrect: the "Latest" field is set to false by default but it should not. Incoming PR

**Pull request merged**